### PR TITLE
Documentation and renaming of "blankies" into somethign a bit easier to ...

### PR DESCRIPTION
**Pull request by jsuereth**
...find.
- Rename SplitExpression\* to `SbtParser` denoting that is parses .sbt files
- Adds a few todos
- Document APIs for internal usage.

Review by @ajozwik  cc> @eed3si9n

### CODE REVIEW

File: main/src/main/scala/sbt/EvaluateConfigurations.scala [hunk 1]
```diff
@@ -218,7 +218,9 @@ object EvaluateConfigurations {
    */
   private[sbt] def splitExpressions(file: File, lines: Seq[String]): (Seq[(String, Int)], Seq[(String, LineRange)]) =
```
Comment (ajozwik) [inline main/src/main/scala/sbt/EvaluateConfigurations.scala:219]: Why we need lines? File is enough - we can read in `SplitExpressionsNoBlankies`. But I think it is for future changes.

**Comment by ajozwik** [comment]
For current changes - no comments. For future - we want to have sbt file - the lines are deprecated.

File: main/src/main/scala/sbt/EvaluateConfigurations.scala (hunk 1 above)
Comment (jsuereth) [inline main/src/main/scala/sbt/EvaluateConfigurations.scala:219]: I agree, sort of.   It's useful to have a method which just takes the raw streams or "virtual file" so we can do in-memory settings (like `set` command in sbt), where multiple settings could be in a string.   Right now we do not do that so you are correct.

I agree we should work on simplifying API now that the feature is pretty stable.   I can add a TODO here about removing the need for lines and reading in splitExpressions.  For now, as it's a private API, it's ok to refactor this at any time.  I"m more concerned with the exposed public APIs _and_ ensuring this area is documented for the next user group that wants to work on it :)

Comment (ajozwik) [inline main/src/main/scala/sbt/EvaluateConfigurations.scala:219]: If I find some time - I can fight with it. But I not promise.
Thank you for your help.

**Pull request merged**